src/app/server-cluster/ServerClusterInterfaceRegistry.h

```diff
@@ -0,0 +1,99 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/ConcreteClusterPath.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+namespace app {
+
+/// Maintains mapping clusters registered on specific endpoints.
+class ServerClusterInterfaceRegistry
+{
+public:
+    /// Associate a specific interface for the given endpoint.
+    ///
+    /// There MUST be a single cluster registration for any given endpoint
+    /// and Register will error out if an existing cluster interface `cluster` already
+    /// exists on the given endpoint for the given clusterid.
+    ///
+    /// Registrations need a valid endpointid and cluster MUST return a valid clusterid.
+    CHIP_ERROR Register(EndpointId endpointId, ServerClusterInterface * cluster);
```

Comment:
I recommend adding [[nodiscard]] to this method. When writing #37078 the issues with the FabricSync tests would've been much easier to diagnose if the CHI registry required the caller to do something with the return value.